routes/provisionSession.ts

```diff
@@ -1,143 +1,128 @@
-import type { Hex } from '@noble/curves/abstract/utils'
-
-export interface AuthReq {
-  deviceId: String
-  sessionId: String
-  siweMsg: {
-    custodyAddress: Hex
-    message: string
-    signature: Uint8Array
-  }
-}
 
 app.post('/provisionSession', async (c) => {
   try {
     const { deviceId, sessionId, siweMsg } = await c.req.json()
 
-    console.log(" received ", deviceId, sessionId, siweMsg)
+    console.log('Received', deviceId, sessionId, siweMsg)
 
-    if (!deviceId || !siweMsg) {
-      return c.json({ success: false, message: 'Missing parameters' }, 400)
+    if (!siweMsg && !sessionId) {
+      return c.json(
+        { success: false, message: 'Missing SIWE message or session ID' },
+        400,
+      )
     }
+    // double check user id type
+    let userId: number
 
-    const { message, signature } = siweMsg
+    let newDeviceId = deviceId
 
-    const selectDeviceQuery = `SELECT userid, deviceid FROM public.hashes WHERE deviceid = $1`
-    const deviceResult = await writeClient.query(selectDeviceQuery, [deviceId])
-    console.log({deviceResult})
-
-    let userId
-
-    if (deviceResult.rows.length > 0) {
-      console.log('Device exists in hashes table **-**')
-      console.log(deviceResult.rows[0])
-
-      if (sessionId) {
-        const { session } = await lucia.validateSession(sessionId)
-        console.log({session})
-
-        if (!session) {
-          return c.json({ success: false, message: 'Invalid session' }, 404)
-        }
-
-        return c.json({
-          success: true,
-          userId: deviceResult.rows[0].userid,
-          sessionId: session.id,
-          deviceId: deviceResult.rows[0].deviceid,
-        })
-
-      } else {
-        // verify message 
-        const isValid = verifyMessage(
-          message,
-          signature,
-          Buffer.from(publicKey).toString('hex'),
+    // Case 1: No device ID (New User)
+    if (!deviceId) {
+      if (!siweMsg) {
+        return c.json(
+          { success: false, message: 'Missing SIWE message for new user' },
+          400,
         )
-
-        if (!isValid) {
-          return c.json({ success: false, message: 'Invalid signature' }, 400)
-        }
-
-        userId = deviceResult.rows[0].userid
-
-        const expiresAt = new Date(Date.now() + 2 * 7 * 24 * 60 * 60 * 1000)
-        const created = new Date(Date.now())
-
-        const session = await lucia.createSession(userId.toString(), {
-          userId: userId,
-          deviceId: deviceId,
-          expiresAt,
-          created,
-        })
-
-        const sessionCookie = lucia.createSessionCookie(session.id)
-        c.header('Set-Cookie', sessionCookie.serialize(), { append: true })
-
-        return c.json({
-          success: true,
-          userId,
-          sessionId: session.id,
-          deviceId: deviceId,
-        })
       }
-    } else {
-      console.log('Device does not exist in hashes table')
+
+      const { message, signature } = siweMsg
```

Comment:
shouldnt the verifyMessage check be happening in the lucia.validateSession helper? isnt that where we override the logic with whatever checks we want to exist between the requst for a session and the provision of one? i could also easily not understand how lucia works